lib/ansible/modules/windows/win_scheduled_task.ps1

```diff
@@ -42,18 +43,26 @@ $state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "prese
 # Vars conditionally required
 $present = $state -eq "present"
 $executable = Get-AnsibleParam -obj $params -name "executable" -type "str" -aliases "execute" -failifempty $present
-$frequency = Get-AnsibleParam -obj $params -name "frequency" -type "str" -validateset "once","daily","weekly" -failifempty $present
+$frequency = Get-AnsibleParam -obj $params -name "frequency" -type "str" -validateset "once","interval","hourly","daily","weekly" -failifempty $present
 $time = Get-AnsibleParam -obj $params -name "time" -type "str" -failifempty $present
 
 # TODO: We should default to the current user
 $user = Get-AnsibleParam -obj $params -name "user" -type "str" -failifempty $present
 
-$weekly = $frequency -eq "weekly"
-$days_of_week = Get-AnsibleParam -obj $params -name "days_of_week" -type "str" -failifempty $weekly
+$frequency_weekly = $frequency -eq "weekly"
+$days_of_week = Get-AnsibleParam -obj $params -name "days_of_week" -type "str" -failifempty $frequency_weekly
 
+$frequency_interval = $frequency -eq "interval"
+$interval = Get-AnsibleParam $params -name "interval" -type "int" -failifempty $frequency_interval
+$interval_unit = Get-AnsibleParam $params -name "interval_unit" -type "str" -validateset "seconds","minutes","hours","days" -failifempty $frequency_interval
 
 try {
-    $task = Get-ScheduledTask -TaskPath "$path" | Where-Object {$_.TaskName -eq "$name"}
+    try {
+        $task = Get-ScheduledTask -TaskPath "$path" -TaskName "$name"
```

Comment:
This needs to be wrapped in a try statement because the script fails when the task doesn't exist, at least on Windows Server 2012R2.